packages/core/src/abstract-wallet-manager.ts

```diff
@@ -0,0 +1,127 @@
+import { EthereumRpcFeeProvider } from '@liquality/ethereum-rpc-fee-provider'
+import { EthereumGasNowFeeProvider } from '@liquality/ethereum-gas-now-fee-provider'
+import { EthereumRpcProvider } from '@liquality/ethereum-rpc-provider'
+import { EthereumJsWalletProvider } from '@liquality/ethereum-js-wallet-provider'
+
+const BITCOIN_FEE_API_URL = 'https://liquality.io/swap/mempool/v1/fees/recommended'
+
+/**
+ * A class that contains functionality that could be shared by the different chains.
+ * The idea is to use strategy pattern to handle logic related to different chains
+ */
+export default class AbstractWalletManager {
+  cryptoassets: any
+  storageManager: StorageManagerI<StateType> | undefined
+  //TODO we need to support other chains as well
+  client: Client | undefined
+
+  protected getNextAccountColor(chain: ChainId, index: number): string {
+    const defaultColor = chainDefaultColors[chain]!
+    const defaultIndex = accountColors.findIndex((c) => c === defaultColor)
+    if (defaultIndex === -1) {
+      return defaultColor
+    }
+    const finalIndex = index + defaultIndex
+    if (finalIndex >= accountColors.length) {
+      return accountColors[defaultIndex]
+    }
+    return accountColors[finalIndex]
+  }
+
+  // Derivation paths calculation
+  protected getBitcoinDerivationPath(coinType: string, index: number) {
+    const BTC_ADDRESS_TYPE_TO_PREFIX = {
+      legacy: 44,
+      'p2sh-segwit': 49,
+      bech32: 84
+    }
+    return `${BTC_ADDRESS_TYPE_TO_PREFIX[bitcoin.AddressType.BECH32]}'/${coinType}'/${index}'`
+  }
+
+  protected getEthereumBasedDerivationPath = (coinType: string, index: number) => `m/44'/${coinType}'/${index}'/0/0`
+
+  protected calculateDerivationPaths(chainId: ChainId): (network: NetworkEnum, index: number) => string | undefined {
+    if (chainId === ChainId.Bitcoin) {
+      return (network: NetworkEnum, index: number) => {
+        const bitcoinNetwork = ChainNetworks[ChainId.Bitcoin]![network]
+        return this.getBitcoinDerivationPath(bitcoinNetwork.coinType, index)
+      }
+    }
+    return (network: NetworkEnum, index: number) => {
+      const ethNetwork = ChainNetworks[ChainId.Ethereum]![network]
+      return this.getEthereumBasedDerivationPath(ethNetwork.coinType, index)
+    }
+  }
+
+  protected createBtcClient(network: NetworkEnum, mnemonic: string, accountType: string, derivationPath: string) {
+    const isTestnet = network === NetworkEnum.Testnet
+    const bitcoinNetwork = ChainNetworks[ChainId.Bitcoin]![network]
+    const esploraApi = config.exploraApis[network]
+    const batchEsploraApi = config.batchEsploraApis[network]
+
+    const btcClient = new Client()
+    btcClient.addProvider(
+      new BitcoinEsploraBatchApiProvider({
+        batchUrl: batchEsploraApi,
+        url: esploraApi,
+        network: bitcoinNetwork as BitcoinNetwork,
+        numberOfBlockConfirmation: 2
+      })
+    )
+
+    btcClient.addProvider(
+      // @ts-expect-error FIXME
```

Comment:
💣

Reply:
Yes. This was put there as placeholder. Eventually, we need to have different implementations of the wallet based on the chain and use them through a strategy pattern. Or another pattern that helps us achieve the flexibility and modularity we need.